### PR TITLE
Cleaned up delay test initialization

### PR DESCRIPTION
**Pull request by evildour**
I didn't go with the attribute route because I think this makes things a bit cleaner for the test method and init method declarations. Happy to change it though if you think the attributes are better for this or other reasons.

### CODE REVIEW

File: apis/Google.Storage.V1/Google.Storage.V1.IntegrationTests/UrlSignerTest.cs [hunk 1]
```diff
@@ -40,6 +42,25 @@ public UrlSignerTest(StorageFixture fixture)
             _fixture.RegisterDelayTests(this);
         }
 
+        /// <summary>
+        /// Gets the name of the delay test associated with the specified delay test initialization method.
+        /// </summary>
+        private static string GetDelayTestName([CallerMemberName] string initMethodName = null)
+        {
+            Debug.Assert(initMethodName.EndsWith(StorageFixture.DelayTestSuffix),
```
Comment (evildour) [inline apis/Google.Storage.V1/Google.Storage.V1.IntegrationTests/UrlSignerTest.cs]: Done. Yes, that's probably true, but I usually like to make things go bang as early as possible so someone debugging in the future doesn't need to backtrack and find out where things went wrong. I'm assuming you want me to change all Debug.Asserts here to GaxPreconditions.CheckState, but let me know if I misunderstood.

Comment (jskeet) [inline apis/Google.Storage.V1/Google.Storage.V1.IntegrationTests/UrlSignerTest.cs]: Either CheckArgument (for the first one) or CheckState (for the others). Whatever makes the most sense, although it frankly doesn't matter too much, so long as it goes bang :)

File: apis/Google.Storage.V1/Google.Storage.V1.IntegrationTests/UrlSignerTest.cs [hunk 2]
```diff
@@ -54,7 +75,7 @@ private void DeleteTest_InitDelayTest()
             var name = GenerateName();
             string url = null;
 
-            _fixture.RegisterDelayTest(GetTestName(nameof(DeleteTest)), _duration,
+            _fixture.RegisterDelayTest(GetDelayTestName(), _duration,
```
Comment (evildour) [inline apis/Google.Storage.V1/Google.Storage.V1.IntegrationTests/UrlSignerTest.cs]: Good idea. Done. I did need to add some 'global' state to the test fixture so the register calls don't need to pass in the declaring class name though, but I don't think its too bad.

**Comment by evildour** [comment]
PTAL

**Comment by evildour** [comment]
Actually, nothing else about registering delay tests is thread safe, so I'm just going to remove the attribute.

**Comment by evildour** [comment]
(rebased)

**Pull request merged**